src/metabase/lib/schema/util.cljc

```diff
@@ -84,13 +84,17 @@
 (mu/defn mbql-clause-distinct-key
   "For deduplicating MBQL clauses: keep just the keys in options that are essential to distinguish one clause from
   another. Removes namespaced keywords and type information keys like `:base-type`."
-  [[tag opts & children]]
-  (into [tag
-         (opts-distinct-key opts)]
-        (map (fn [child]
-               (cond-> child
-                 (mbql-clause? child) mbql-clause-distinct-key)))
-        children))
+  [clause]
```

Comment:
This code is significantly more verbose than the original code, so it's worth adding a comment explaining that you're doing it this way for performance reasons. Otherwise there's a chance someone will come back and clean it up and write something closer to the original in the future